proxy/src/main/java/pt/webdetails/cda/CdaUtils.java

```diff
@@ -0,0 +1,611 @@
+
+  private static String CDA_HOST = System.getProperty( "cda.hostname" );
+  private static String CDA_PORT = System.getProperty( "cda.port" );
+  private static String USER = System.getProperty( "repos.user" );
+  private static String PASS = System.getProperty( "repos.password" );
+  private String url_do_query = "http://" + CDA_HOST + ":"
+      + CDA_PORT + "/cxf/cda/cda/api/utils/doQuery";
+
+  public CdaUtils() {
+    String serverFlag = System.getProperty( "cda.to_server" );
+    if ( serverFlag != null && serverFlag.equals( "1" ) ) {
+      url_do_query = "http://" + CDA_HOST + ":" + CDA_PORT + "/pentaho/plugin/cda/api/doQuery";
+    }
+  }
+
+  @POST
+  @Path( "/doQuery" )
+  @Consumes( APPLICATION_FORM_URLENCODED )
+  @Produces( { MimeTypes.JSON, MimeTypes.XML, MimeTypes.CSV, MimeTypes.XLS, MimeTypes.PLAIN_TEXT, MimeTypes.HTML } )
+  public StreamingOutput doQueryPost( MultivaluedMap<String, String> formParams,
+      @Context HttpServletRequest servletRequest,
+      @Context HttpServletResponse servletResponse ) throws WebApplicationException {
+
+    //proxy request to the real CDA Endpoint
+
+    String url = url_do_query + servletRequest.getQueryString();
+
+    //Init
+    ClientConfig clientConfig = new DefaultClientConfig();
+    clientConfig.getFeatures().put( JSONConfiguration.FEATURE_POJO_MAPPING, Boolean.TRUE );
+    Client client = Client.create( clientConfig );
+    client.addFilter( new HTTPBasicAuthFilter( USER, PASS ) );
+
+    //Invoke Rest endpoint with same params, converting POST into a GET request
+    WebResource webResource = client.resource( url );
+
+    // add parameters
+    Iterator<String> it = formParams.keySet().iterator();
+    String key;
+    while ( it.hasNext() ) {
+      key = it.next();
+      webResource = webResource.queryParam( key, formParams.get( key ).get( 0 ) );
+    }
+
+    try {
+      ClientResponse response = webResource.type( MediaType.APPLICATION_OCTET_STREAM_TYPE )
+        .get( ClientResponse.class );
+
+      if ( response.getStatus() == 200 ) {
+        InputStream in = response.getEntityInputStream();
+
+        return new StreamingOutput() {
+          @Override
+          public void write( OutputStream outputStream ) throws IOException, WebApplicationException {
+            try {
+              IOUtils.copy( in, outputStream );
+            } finally {
+              IOUtils.closeQuietly( in );
+            }
+          }
+        };
+      }
+    } catch ( Exception ex ) {
+      logger.fatal( ex );
+    } finally {
+      client.destroy();
+    }
+
+    return null;
+  }
+
+  /**
+   * For CDE discovery
+   * Used by CPF PluginCall, when CDE Editor opens in Pentaho Server the first time
+   */
+  @GET
+  @Path( "/listDataAccessTypes" )
+  @Produces( APPLICATION_JSON )
+  @Consumes( { APPLICATION_XML, APPLICATION_JSON } )
+  public String listDataAccessTypes( @DefaultValue( "false" ) @QueryParam( "refreshCache" ) Boolean refreshCache )
```

Comment:
Not sure if instead we could call `/listDataAccessTypes` on the real CDA. I think currently it is also an hardcoded string, but that way we would only maintain one.